modules/apps/27-interchain-accounts/controller/keeper/handshake.go

```diff
@@ -74,18 +75,25 @@ func (k Keeper) OnChanOpenAck(
 		return sdkerrors.Wrapf(porttypes.ErrInvalidPort, "portID cannot be host chain port ID: %s", icatypes.PortID)
 	}
 
-	if err := icatypes.ValidateVersion(counterpartyVersion); err != nil {
-		return sdkerrors.Wrap(err, "counterparty version validation failed")
+	if !strings.HasPrefix(portID, icatypes.PortPrefix) {
+		return sdkerrors.Wrapf(porttypes.ErrInvalidPort, "controller port %s does not contain expected prefix %s", portID, icatypes.PortPrefix)
 	}
 
-	k.SetActiveChannelID(ctx, portID, channelID)
+	var metadata icatypes.Metadata
```

Comment:
Would there be value in adding a `validateMetadata` fn and using it here? Also wondering should we use a constructor fn for creating (add to types)

Reply:
I thought about this myself when I was implementing it, but I came to the conclusion that it wouldn't be so reusable as we need to carry out metadata validation in `OnChanOpenAck` as well, and we do not have a ref to connectionHops at that point in the handshake flow.

Reply:
I think we still need to validate connection sequences here? It is possible the host chain chooses a bad metadata, followup issue

Reply:
Sure, I can address this in a followup PR, with the moving of `validateConnectionParams` to `types`.

I assume retrieving the channel from state using channelKeeper, and passing along it's `connectionHops` to `ValidateConnectionParams` should do the trick 👍

Reply:
Figured it made more sense to move this to a `ValidateMetadata` function in the end, since `connectionHops` can be retrieved from the channel stored in state for `ChanOpenAck`. I can open a follow up PR with [the changes](https://github.com/cosmos/ibc-go/compare/damian/615-refactor-ica-connection-ids...damian/615-followup?expand=1) when this is merged.